packages/app/models/token/types.ts

```diff
@@ -1,10 +1,19 @@
-export interface Token {
+import { TokenSubgraph } from "@stackly/sdk";
+
+type TokenWithoutIdAndAddress = Omit<TokenSubgraph, "id" | "address">;
+
+interface TokenWithAddress extends TokenWithoutIdAndAddress {
   address: string;
-  decimals: number;
-  name: string;
-  symbol: string;
 }
 
+interface TokenWithId extends TokenWithoutIdAndAddress {
+  address: string;
+}
```

Comment:
Both Types have `address`

Reply:
This looks cleaner

```suggestion
type TokenWithAddress = Omit<TokenSubgraph, "id">;
type TokenWithId = Omit<TokenSubgraph, "address">;
```

Reply:
Somehow get an error trying this.

Reply:
> Somehow get an error trying this.

I think it's working because all tokens have address. Event `TokenWithId`. Also there is no token with id and without address now that I'm thinking. So I think the join should be `TokenWithAddress | TokenSubgraph`

Reply:
Isn't there Tokens with only `address` from `tokenlists`

Reply:
Yeah, but there aren't tokens with only `id`. They came from subgraph but are now fixed.

Reply:
true, makes sense, good catch.

fixed here: https://github.com/SwaprHQ/stackly-ui/pull/40/commits/ea5bde9b1f505ec22e40df467449e7c53e76d567